### PR TITLE
Fix minesweeper rep earned message

### PR DESCRIPTION
**Pull request by jwill0213**
Fix issue with passing context with data into translator translate function.

Add test file for translator.

### CODE REVIEW

File: tests/charbot/test_translator.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+def locale() -> Locale:
+    return Locale.american_english
+
+
+@pytest.mark.asyncio
+async def test_translate_with_data(translator: Translator, locale: Locale):
+    context = TranslationContext(TranslationContextLocation.other, data={"awarded": 5})
+    key = locale_str("minesweeper-win-description")
+    result = await translator.translate(key, locale, context)
+    assert "You gained 5 points." in result, "Should say awarded 5 points"
```
Comment (Bluesy1) [inline tests/charbot/test_translator.py:25]:
```suggestion
    assert result is not None, "Result should not be None"
    assert "You gained 5 points." in result, "Should say awarded 5 points"
```

**Review by Bluesy1 — CHANGES_REQUESTED** [review]
One small typing nitpick, otherwise LGTM!

**Comment by jwill0213** [comment]
Removed match case for translator.py. Pyright was being very picky with the types and just made a mess to use. Would have made the code a lot less readable.

**Comment by Bluesy1** [comment]
Hopefully pyright behaves ... I'll figure out a solution eventually to use match-case

**Pull request merged**